sdcm/nemesis.py

```diff
@@ -5246,6 +5246,38 @@ def disrupt_grow_shrink_zero_nodes(self):
         znode = random.choice([node for node in self.cluster.zero_nodes if node.dc_idx == self.target_node.dc_idx])
         self.decommission_nodes(nodes=[znode])
 
+    @target_all_nodes
+    def disrupt_serial_restart_elected_topology_coordinator(self):
+        """ Serial restart of elected topology coordinator node,
+        should trigger new coordinator node election
+        """
+        if not self.target_node.raft.is_consistent_topology_changes_enabled:
+            raise UnsupportedNemesis("Consistent topology changes feature is disabled")
+
+        self.use_nemesis_seed()
+        num_of_restarts = random.randint(1, len(self.cluster.nodes))
+        self.log.debug("Number of serial restart of topology coordinator: %s", num_of_restarts)
+        election_wait_timeout = random.choice([1, 5, 10, 15])
+        self.log.debug("Wait new topology coordinator election timeout: %s", election_wait_timeout)
+        for _ in range(num_of_restarts):
+            coordinator_node = get_topology_coordinator_node(self.target_node)
+            self.log.debug("Coordinator node: %s, %s", coordinator_node, coordinator_node.name)
+            self.unset_current_running_nemesis(self.target_node)
```

Comment:
@soyacz , i do that because target_node could be the topology coordinator, and nemesis will be aborted at first run. From other side, The target_node is already locked for nemesis and it could be safely used to find coordinator node.
in other hand there will be too many nested with .. statements

Reply:
as for verification nodes we usually don't lock it, so no need to do it here too I suppose.
Anyway if you unset running nemesis from self.target_node for a while, it may be taken in that time by parallel nemesis, also self.target_node may happen to become coordinator, so it will fail in that case I suppose.
Better just select verification node randomly out of all nodes that are not currently running nemesis.

Reply:
I think that's a mistake we don't lock them.

other nemesis can take them down, and we'll fail while doing any action with them,

every time we work with a node in nemesis code, the node should be marked, and when we mark, we should hold the lock.

Reply:
in this particular case - disruptive nemesis - no other nemesis should take it down (as we don't run 2 disruptive in parallel) and verification is non disruptive (and should be short). We may lock it for a short a while just to give a good example.
Anyway currently, self.target_node may be down due to this nemesis itself - so this must be changed.